runtime/src/builtins.rs

```diff
@@ -123,6 +123,24 @@ pub enum BuiltinFeatureTransition {
     },
 }
 
+// https://github.com/solana-labs/solana/pull/23233 added `BuiltinFeatureTransition`
+// to `Bank` which triggers https://github.com/rust-lang/rust/issues/92987 while
+// attempting to resolve `Sync` on `BankRc` in `AccountsBackgroundService::new` ala,
+//
+// query stack during panic:
+// #0 [evaluate_obligation] evaluating trait selection obligation `bank::BankRc: core::marker::Sync`
+// #1 [typeck] type-checking `accounts_background_service::<impl at runtime/src/accounts_background_service.rs:358:1: 520:2>::new`
+// #2 [typeck_item_bodies] type-checking all item bodies
+// #3 [analysis] running analysis passes on this crate
+// end of query stack
+//
+// Yoloing a `Sync` onto it avoids the auto trait evaluation and thus the ICE.
```

Comment:
Which is safe because the compiler was cool with the same types in tuple form before #23233, right?

Reply:
True, but how about adding some extra checks to ensure that our assumptions aren't broken in the future. Here are my suggested changes: https://github.com/t-nelson/solana/pull/32

Reply:
@jstarry what assumptions are you referring to that may be broken in the future?

Reply:
@jackcmay the unsafe impl can introduce undefined behavior because the compiler won't complain if they are added for types which are not thread-safe.